diff --git a/src/util.cpp b/src/util.cpp
index 78c353d..0dc5b01 100644
--- a/src/util.cpp
+++ b/src/util.cpp
@@ -487,7 +487,7 @@ boost::filesystem::path GetDefaultDataDir()
     return GetSpecialFolderPath(CSIDL_APPDATA) / "Bitcoin";
 #else
     fs::path pathRet;
-    char* pszHome = getenv("HOME");
+    char* pszHome = getenv("SNAP_USER_COMMON");
     if (pszHome == NULL || strlen(pszHome) == 0)
         pathRet = fs::path("/");
     else
