diff --git a/namecoin-qt.pro b/namecoin-qt.pro
index 7b712b7..9b4f259 100644
--- a/namecoin-qt.pro
+++ b/namecoin-qt.pro
@@ -2,12 +2,15 @@ TEMPLATE = app
 TARGET = namecoin-qt
 macx:TARGET = "Namecoin-Qt"
 VERSION = 0.3.80
-INCLUDEPATH += src src/json src/qt
+INCLUDEPATH += src src/json src/qt $(SNAPCRAFT_STAGE)/include
 QT += core gui network
 greaterThan(QT_MAJOR_VERSION, 4): QT += widgets
 DEFINES += GUI QT_GUI BOOST_THREAD_USE_LIB BOOST_SPIRIT_THREADSAFE
 CONFIG += no_include_pwd
 CONFIG += thread
+BDB_LIB_SUFFIX = -4.8
+
+LIBS += -L$(SNAPCRAFT_STAGE)/lib
 
 DEFINES += NOPCH FOURWAYSSE2 USE_SSL
 
