diff --git a/src/qt/guiutil.cpp b/src/qt/guiutil.cpp
index 876b8b1..205f767 100644
--- a/src/qt/guiutil.cpp
+++ b/src/qt/guiutil.cpp
@@ -704,7 +704,7 @@ boost::filesystem::path static GetAutostartDir()
 
     char* pszConfigHome = getenv("XDG_CONFIG_HOME");
     if (pszConfigHome) return fs::path(pszConfigHome) / "autostart";
-    char* pszHome = getenv("HOME");
+    char* pszHome = getenv("SNAP_USER_COMMON");
     if (pszHome) return fs::path(pszHome) / ".config" / "autostart";
     return fs::path();
 }
diff --git a/src/util.cpp b/src/util.cpp
index e5697a6..baf6b90 100644
--- a/src/util.cpp
+++ b/src/util.cpp
@@ -464,7 +464,7 @@ boost::filesystem::path GetDefaultDataDir()
     return GetSpecialFolderPath(CSIDL_APPDATA) / "Namecoin";
 #else
     fs::path pathRet;
-    char* pszHome = getenv("HOME");
+    char* pszHome = getenv("SNAP_USER_COMMON");
     if (pszHome == NULL || strlen(pszHome) == 0)
         pathRet = fs::path("/");
     else
